anyric/super-record
# #167261155 Add User Role Model with Permissions

#### What does this PR do?
This PR add user roles with permissions

#### Description of Task to be completed?
- Add user roles model for managing roles
- Implement a roles view interface
- Implement an edit endpoint for editing existing user roles
- Implement a delete endpoint for deleting user roles

#### How should this be manually tested?
- Clone this branch
- In the root directory run `python manage.py runserver`
- Login to the system
- Navigate to `http://127.0.0.1:5000/setting` and select `Roles` menu
- Click on the `New Role` button and fill the form
- Confirm the action from the modal form to complete

#### Any background context you want to provide?
Currently there are no user roles define in the system. There should be roles model for managing different user groups by assigning and revoking the roles for a group.

#### What are the relevant pivotal tracker stories?
[#167261155](https://www.pivotaltracker.com/story/show/167261155)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/29800415/61861613-fba97f80-aed4-11e9-9e35-93cd1503004f.png)

![image](https://user-images.githubusercontent.com/29800415/61861639-049a5100-aed5-11e9-9ddf-3a0169bc1d3e.png)

![image](https://user-images.githubusercontent.com/29800415/61861652-0a903200-aed5-11e9-91ad-68dcbfee49a1.png)

![image](https://user-images.githubusercontent.com/29800415/61874880-20602000-aef2-11e9-9a85-18e9740df96b.png)

#### Questions:
N/A
